File: Array/TwoSum/TwoSum_Fomagran.js
class Enum {
    constructor(index, value) {
        this.index = index
        this.value = value
    }
}

var twoSum = function (nums, target) {
    const enums = []
    nums.forEach((value, index) => {
        enums.push(new Enum(index, value))
    })

    const sorted = enums.sort(function (a, b) {
        return a.value - b.value
    })

    for (let i = sorted.length - 1; i >= 0; i--) {
        let large = sorted[i]
        for (let j = 0; i < sorted.length; j++) {
            let small = sorted[j]
            console.log(small.value, large.value)
            if (large.value + small.value == target) {
                return [small.index, large.index]
            }
            if (large.value + small.value > target) {
                break
            }
        }
    };
}

File: Array/TwoSum/TwoSum_Fomagran.swift
func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    let map = nums.enumerated().map{$0}
    let sorted = map.sorted{$0.element <  $1.element}
    let reversed = sorted.reversed()
    for large in reversed {
        for small in sorted {
            if large.element + small.element == target {
                return [small.offset,large.offset]
            }
            if large.element + small.element > target {
                break
            }
        }
    }
    return []
}